mPower2/mPower2/HistoryFeedTableViewController.swift

```diff
@@ -0,0 +1,280 @@
+            }
+        default:
+            break;
+        }
+    }
+    
+    func controller(_ controller: NSFetchedResultsController<NSFetchRequestResult>, didChange sectionInfo: NSFetchedResultsSectionInfo, atSectionIndex sectionIndex: Int, for type: NSFetchedResultsChangeType) {
+        switch (type) {
+        case .insert:
+            tableView.insertSections([sectionIndex], with: .fade)
+        case .delete:
+            tableView.deleteSections([sectionIndex], with: .fade)
+        default:
+            break;
+        }
+    }
+    
+    func controllerDidChangeContent(_ controller: NSFetchedResultsController<NSFetchRequestResult>) {
+        tableView.endUpdates()
+    }
+}
+
+class HistoryFeedBaseTableViewCell : UITableViewCell {
+    func configure(for item: HistoryItem, indexPath: IndexPath) {
+    }
+}
+
+class HistoryFeedTableViewCell : HistoryFeedBaseTableViewCell {
+    
+    @IBOutlet weak var taskIcon: UIImageView!
+    @IBOutlet weak var taskTitleLabel: UILabel!
+    @IBOutlet weak var dotView: UIView!
+    @IBOutlet weak var taskDetailLabel: UILabel!
+    @IBOutlet weak var dotWidth: NSLayoutConstraint!
+    
+    override func configure(for item: HistoryItem, indexPath: IndexPath) {
+        
+        let designSystem = RSDDesignSystem()
+        let background = designSystem.colorRules.backgroundLight
+        
+        taskIcon.image = item.image
+        taskTitleLabel.text = item.title
+        taskTitleLabel.textColor = designSystem.colorRules.textColor(on: background, for: .mediumHeader)
+        taskDetailLabel.textColor = designSystem.colorRules.textColor(on: background, for: .bodyDetail)
+        
+        if let symptomsItem = item as? SymptomHistoryItem,
+            let severity = SBASymptomSeverityLevel(rawValue: Int(symptomsItem.severityLevel)) {
+            // Show the dot and the detail label
+            dotWidth.constant = 18
+            dotView.layer.cornerRadius = dotWidth.constant / 2
+            dotView.backgroundColor = designSystem.colorRules.severityColorScale.stroke(for: severity.rawValue, isSelected: true)
+            switch severity {
+            case .none:
+                taskDetailLabel.text = Localization.localizedString("SYMPTOM_SEVERITY_NONE")
+            case .mild:
+                taskDetailLabel.text = Localization.localizedString("SYMPTOM_SEVERITY_MILD")
+            case .moderate:
+                taskDetailLabel.text = Localization.localizedString("SYMPTOM_SEVERITY_MODERATE")
+            case .severe:
+                taskDetailLabel.text = Localization.localizedString("SYMPTOM_SEVERITY_SEVERE")
+            }
+        }
+        else {
+            // If not a symptom, then hide the dot.
+            dotView.backgroundColor = UIColor.clear
+            dotWidth.constant = 0
+            
+            if let tapItem = item as? TapHistoryItem {
+                // syoung 07/18/2019 This will work for English. Not sure how well this formatting
```

Comment:
Maybe make this a TODO: so it's a teeny bit easier to find later? Maybe the answer will turn out to be as simple as using a localized string for the separator, if we're very very lucky.